modules/gcp-gke/node_pool_tags.tf

```diff
@@ -0,0 +1,18 @@
+###
+# This file contains no business logic.
+# The solely goal is to provide a way to read out the tags of the node pool, which include the ones we apply with this component + the ones Generated by Google.
+###
+
+locals {
+  # includes the by GCP-managed tags
+  primary_node_pool_all_tags = sort(data.google_compute_instance.primary_node_pool.tags)
+}
+
+data "google_compute_instance_group" "primary_node_pool" {
+  depends_on = [google_container_node_pool.primary_node_pool]
+  self_link  = element(google_container_node_pool.primary_node_pool.managed_instance_group_urls, 0)
+}
+
+data "google_compute_instance" "primary_node_pool" {
+  self_link = sort(data.google_compute_instance_group.primary_node_pool.instances)[0]
```

Comment:
Why are we sorting here?

Reply:
Short answer: No reason.

Long answer: The examples where we saw they were using this way of reading out the tags had this `sort` included. I can imagine one feels more confident as in every run it will use the same instance to read the tags? I can also imagine it mitigates drift, as probably picking up always the first instance gives always the same tags? Not sure.

Reply:
I figured out: `data.google_compute_instance_group.primary_node_pool.instances` is not a list, it is a set. So one may not write `data.google_compute_instance_group.primary_node_pool.instances[0]`. Using `tolist` instead of sort, but seems like the same effort.